tinygrad/codegen/linearizer.py

```diff
@@ -113,7 +115,7 @@ def rename_var(v: VariableOrNum, expr: str): return v if isinstance(v, NumNode)
           rendered_idx = idx.render(self.render_ops, self)
           valid_tuple = (valid.render(self.render_ops, self), self.const(invalid_value, localtype)) if valid.min == 0 else tuple()
           self.load_cache[key] = self.uop(UOps.LOAD, localtype, (buf_uop, rendered_idx) + valid_tuple + ((barrier,) if barrier else ()))
-      ret.append(self.uop(UOps.GEP, localtype.scalar(), (self.load_cache[key],), rep_idx[dim]) if dim is not None else self.load_cache[key])
+      ret.append(self.uop(UOps.GEP, localtype.scalar() if not (self.opts.device == "HIP" and localtype == dtypes.half.vec(2)) else dtypes.uint16, (self.load_cache[key],), rep_idx[dim]) if dim is not None else self.load_cache[key]) # In HIP, the elements within half2 are uint16
```

Comment:
There should not be device specific hacks in the linearizer

Reply:
This is what it takes https://github.com/tinygrad/tinygrad/pull/2264/commits/7515fb2b4251f6c1908f06511bee2f34ee60a22f :) I'm ok with it i think it's a better place to put it, until we have a uop pre-processor based on device maybe (don't know how that'd look like)